### PR TITLE
Disable submit buttons after form submission on instructorFeedbackEdit page #5582

### LINKED ISSUE #5582: Disable submit buttons after form submission on instructorFeedbackEdit page
In general, we should disable the submit button once the form is submitted. When the site is slow, users may get impatient and click the button multiple times, possibly resulting in incorrect behavior. For example, currently on the feedback edit page, after adding a question, an instructor can press submit multiple times. This actually results in multiple of the same question getting added. When we allow instructors to send reminders, this may also result in multiple emails getting sent. We should see if we can specify this behavior in common.js, otherwise make multiple issues for the different forms.

### PR DESCRIPTION
**Pull request by rclakmal**
Fixes #5582

### CODE REVIEW

File: src/main/webapp/js/instructorFeedbackEdit.js [hunk 1]
```diff
@@ -21,6 +21,19 @@ $(document).ready(function() {
     hideUncommonPanels();
 });
 
+function addLoadingIndicator(button) {
+    currentText = button.text();
+    button.prop('disabled', true);
+    button.append("<img src='/images/ajax-loader.gif'/>");
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorFeedbackEdit.js]: Two issues here:
- Switch the `'` and `"`; we use `'` in JS and `"` in HTML.
- Don't close the tag; we don't close void tags based on our style guide.

File: src/test/resources/pages/instructorFeedbackConstSumOptionQuestionAddSuccess.html [hunk 2]
```diff
@@ -2124,7 +2124,9 @@ <h5 class="col-sm-2">
         </div>
         <div>
           <span class="pull-right">
-            <input class="btn btn-primary" id="button_submit_add" tabindex="9" type="submit" value="Save Question">
+                        <button class="btn btn-primary" id="button_submit_add" tabindex="9" type="submit">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackConstSumOptionQuestionAddSuccess.html]: Weird that this is indented this far off here. Is this generated with God mode?

File: src/main/webapp/WEB-INF/tags/instructor/feedbackEdit/newQuestionForm.tag [hunk 3]
```diff
@@ -119,8 +119,10 @@
           
             <div>
                 <span class="pull-right">
-                    <input id="button_submit_add" class="btn btn-primary"
-                        type="submit" value="Save Question" tabindex="9">
+                    <button id="button_submit_add" class="btn btn-primary"
+                        type="submit" tabindex="9">
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/instructor/feedbackEdit/newQuestionForm.tag]: Both should be fine on the previous line, even though it was line-broken before.

**Comment by wkurniawan07** [comment]
@rclakmal also, any screenshots + any way this feature can be tested?

**Comment by rclakmal** [comment]
@wkurniawan07 Fixed. 

Idea of the original issue was that if a certain user press `Save Question` button multiple times, the post request get submitted multiple times and result in duplicates of the same question. Once user clicks on the button it disables and only enables after process is finished.  (either successful reload or form error)
screen shot 2016-07-18 at 9 36 34 pm

**Comment by damithc** [comment]
From a Ui perspective, it is good if the button size remains the same even when the spinner is showing.
e.g. 
`[Save Question]` 
`[Saving . . .  ]`

File: src/main/webapp/js/instructorFeedbackEdit.js [hunk 4]
```diff
@@ -55,7 +73,11 @@ function readyFeedbackEditPage() {
     });
 
     $('form.form_question').submit(function() {
-        return checkFeedbackQuestion(this);
+        formStatus = checkFeedbackQuestion(this);
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorFeedbackEdit.js]: Is `formStatus` a global variable?

File: src/main/webapp/js/instructorFeedbackEdit.js [hunk 5]
```diff
@@ -21,6 +21,19 @@ $(document).ready(function() {
     hideUncommonPanels();
 });
 
+function addLoadingIndicator(button) {
+    currentText = button.text();
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorFeedbackEdit.js]: Similarly, is `currentText` a global variable?

File: src/test/resources/pages/instructorFeedbackConstSumOptionQuestionAddSuccess.html [hunk 6]
```diff
@@ -2124,7 +2124,9 @@ <h5 class="col-sm-2">
         </div>
         <div>
           <span class="pull-right">
-            <input class="btn btn-primary" id="button_submit_add" tabindex="9" type="submit" value="Save Question">
+              <button class="btn btn-primary" id="button_submit_add" tabindex="9" type="submit">
+                  Save Question
+              </button>
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackConstSumOptionQuestionAddSuccess.html]: I'm quite sure this is not regenerated with God mode. Can you regenerate the HTML files with God mode?

**Comment by rclakmal** [comment]
@wkurniawan07 Thanks for the review. Updated loading UI text as instructed by @damithc . Fixed JS variable issue. Re-generated page contents.

**Comment by wkurniawan07** [comment]
@rclakmal code looks good, can you make a gif screen capture showing the button being disabled after submission?

**Comment by rclakmal** [comment]
cdrppmktoi

here you go @wkurniawan07

**Comment by wkurniawan07** [comment]
Looks good! :+1:

**Comment by wkurniawan07** [comment]
@rclakmal can you update this branch to be merged?

**Pull request merged**